library/Rules/Json.php

```diff
@@ -15,11 +15,12 @@ class Json extends AbstractRule
 {
     public function validate($input)
     {
-        if (is_string($input)
-            && strtolower($input) == 'null') {
-            return true;
+        if (!is_string($input)) {
+            return false;
         }
 
-        return (null !== json_decode($input));
+        $decode = json_decode($input);
+
+        return is_object($decode) || is_array($decode);
```

Comment:
```php
json_decode($input);

return (json_last_error() === JSON_ERROR_NONE);
```

I've found this part of code in many places on the web, telling that would validate a json. but in fact many people has also complained about it exactly because it would also validate only numbers or certain strings.

according to json,org actually, a json is built on TWO data structures, AND, inside these data structures there will be a value, and yes, these values can be:

* string
* number
* object
* array
* true
* false
* null